addons/l10n_ar_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,134 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import logging
+
+from odoo import models, fields, api, Command, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountPaymentRegister(models.TransientModel):
+    _inherit = 'account.payment.register'
+
+    l10n_ar_withholding_ids = fields.One2many(
+        'l10n_ar.payment.register.withholding', 'payment_register_id', string="Withholdings",
+        compute='_compute_withholdings', readonly=False, store=True)
+    l10n_ar_net_amount = fields.Monetary(compute='_compute_l10n_ar_net_amount', readonly=True,  help="Net amount after withholdings")
+
+    @api.depends('line_ids', 'can_group_payments', 'group_payment')
+    def _compute_withholdings(self):
+        supplier_recs = self.filtered(lambda x: x.partner_type == 'supplier' and (not x.can_group_payments or (x.can_group_payments and x.group_payment)))
+        for rec in supplier_recs:
+            taxes = rec._get_withholding_tax()
+            rec.l10n_ar_withholding_ids = [Command.clear()] + [Command.create({'tax_id': x.id, 'base_amount': 0}) for x in taxes]
+        (self - supplier_recs).l10n_ar_withholding_ids = False
+
+    def _compute_amount(self):
+        super()._compute_amount()
+        self.filtered(lambda x: x.l10n_ar_withholding_ids and x.l10n_latam_check_id)._compute_amount_from_latam_check()
+
+    def _compute_amount_from_latam_check(self):
+        for wizard in self.filtered(lambda x: x.l10n_ar_withholding_ids and x.l10n_latam_check_id):
+            simulated_wizard = self.env['account.payment.register'].with_context(active_model='account.move.line', active_ids=wizard.line_ids.ids).new()
+            simulated_wizard.currency_id = wizard.l10n_latam_check_id.currency_id
+            simulated_wizard.payment_date = wizard.payment_date
+            factor = simulated_wizard.amount / simulated_wizard.l10n_ar_net_amount
+            manual_withholding_lines = wizard.l10n_ar_withholding_ids.filtered(lambda x: x.tax_id not in simulated_wizard.l10n_ar_withholding_ids.mapped('tax_id'))
```

Comment:
I wonder if this manual adaptation works well.  If not, we could just let the user adapt the amounts himself and put the net amount in red and not allow him to pay with a wrong amount.